schema/deploy/shipping/views.sql

```diff
@@ -557,4 +557,104 @@ create or replace view shipping.observation_with_presence_absence_result_v2 as
 comment on view shipping.observation_with_presence_absence_result_v2 is
   'Joined view of shipping.incidence_model_observation_v3 and shipping.presence_absence_result_v1';
 
+
+create or replace view shipping.sample_with_best_available_site_details_v1 as
+
+    with specimen_manifest_site_data as (
+        select distinct on (manifest_id)
+            cast(value->>'sample_id' as integer) as sample_id,
+            trim(both ' ' from document->>'swab_site') as swab_site,
+            trim(both ' ' from document->>'sample_origin') as sample_origin
+        from
+            receiving.manifest,
+            jsonb_array_elements(processing_log)
+        where value->'sample_id' is not null
+        order by manifest_id desc
+    ),
+
+    site_details as (
+      select
+          site_id,
+          site.details->>'type' as site_type,
+          site.details->>'category' as site_category
+      from warehouse.site
+    ),
+
+    site_type_to_category_mapping as (
+      select distinct on (details->>'type')
+        details->>'type' as site_type,
+        details->>'category' as site_category
+      from warehouse.site
+      where details->>'category' is not null
+    ),
+
+    normalized_site_types as (
+      select
+        sample_id,
+        sample.identifier as sample,
+        coalesce(site_type,
+          case
+              -- First handle sample_origin
+              when sample_origin = 'es'
+                then 'environmental'
+
+              when sample_origin like '%_retro'
```

Comment:
It'd be a little more robust to match case-insensitively using `ilike` instead of `like`.

Reply:
I addressed this in a new commit.

Reply:
:disappointed: Realizing now that there's no equivalent for `similar to`.